### PR TITLE
[V] Remove complex type fallback for parquet

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?

Remove complex type fallback for parquet 

## How was this patch tested?

leveraging existing UT `GlutenParquetV2SchemaPruningSuite`.

### CODE REVIEW

File: backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala [hunk 1]
```diff
@@ -158,11 +159,12 @@ object VeloxBackendSettings extends BackendSettingsApi {
               CharVarcharUtils.getRawTypeString(metadata) + " not support"
             case StructField(_, TimestampType, _, _) => "TimestampType not support"
           }
-          if (!GlutenConfig.getConf.forceComplexTypeScanFallbackEnabled) {
-            validateTypes(typeValidator)
-          } else {
-            validateTypes(orcTypeValidatorWithComplexTypeFallback)
-          }
+          ValidationResult.succeeded
```
Comment (FelixYBW) [inline backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:162]: @kecookier have you use complex datatype in ORC format?

Comment (yma11) [reply backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:162]: This is for just a try how the change affect ORC related UTs. It turns out timestamp support has result mismatch issue. We may still keep this option for ORC as it's not fully verified as Parquet. For parquet, we have problems in nested struct type support in Gluten and I am looking into that.

Comment (kecookier) [reply backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:162]: > @kecookier have you use complex datatype in ORC format?

Yes, we disable option `forceComplexTypeScanFallbackEnabled`.

Comment (jinchengchenghh) [reply backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala]: Does the timestamp support result match issue involves Velox? Could we create a new issue in Velox to track it?

File: backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala [hunk 2]
```diff
@@ -87,56 +87,8 @@ object VeloxBackendSettings extends BackendSettingsApi {
...
-            "ArrayType as element in ArrayType"
-          // Parquet scan of nested map with struct as key type,
-          // or array type as value type is not supported in Velox.
-          case StructField(_, mapType: MapType, _, _) if mapType.keyType.isInstanceOf[StructType] =>
-            "StructType as Key in MapType"
-          case StructField(_, mapType: MapType, _, _)
-              if mapType.valueType.isInstanceOf[ArrayType] =>
-            "ArrayType as Value in MapType"
-          case StructField(_, TimestampType, _, _)
-              if GlutenConfig.getConf.forceParquetTimestampTypeScanFallbackEnabled =>
```
Comment (zml1206) [inline backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:131]: Reading the int64 timestamp in the parquet file does not seem to be supported yet.
https://github.com/facebookincubator/velox/pull/8325

File: backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala (hunk 1 above)
Comment (yma11) [reply backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala]: Issue [6831](https://github.com/apache/incubator-gluten/issues/6831) is created for track.

File: backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala (hunk 2 above)
Comment (yma11) [reply backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala:143]: Yes. I keep this check and we may have a try by setting `forceParquetTimestampTypeScanFallbackEnabled` to false once the related support merged.

File: backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala (hunk 1 above)
Comment (jinchengchenghh) [reply backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxBackend.scala]: Thanks~

**Comment by yma11** [comment]
@FelixYBW can you help approve this PR? Thanks.

**Review by FelixYBW — APPROVED** [review]

**Comment by FelixYBW** [comment]
We have passed all Spark UTs with complex datatype in parquet.

**Pull request merged**